src/Models/DataObjectTaxonomyTerm.php

```diff
@@ -32,4 +32,23 @@ class DataObjectTaxonomyTerm extends DataObject
     private static $extensions = [
         Versioned::class,
     ];
+
+
+    /**
+     * Though this object is used to link {@see DataObject} and {@see TaxonomyTerm} in a many-many relation, the
+     * object itself is manipulated as a data object and is {@see Versioned} by default, it could be published by its
+     * owner object through the static $owns {@see DataObjectTaxonomiesDataExtension::$owns} defined. On an operation
+     * of deleting this object trigged by 'unlink' on its owner's interface, this object is removed only from its
+     * Staging stage, whereas leaves its published record as an orphaned record. The function is to ensure the object's
+     * published record is also removed to maintains data integrity.
+     */
+    public function onAfterDelete()
```

Comment:
Should we use `onBeforeDelete` instead?

Reply:
That is what I have in the first place, but after testing, it sometimes produces server errors on a specific object and I suspect it is due to some other extensions those objects having and chaining hooks.

Reply:
Is it a big deal to figure out what is going on with onBeforeDelete? It should be safe to unpublished before deleting. Just move the parent call to be the first thing in the method.